src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs

```diff
@@ -0,0 +1,95 @@
+using System.Management.Automation;
+using Docker.DotNet.Models;
+using System.Threading.Tasks;
+using System;
+using System.Linq;
+using Docker.PowerShell.Objects;
+using System.IO;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Docker.PowerShell.Cmdlets
+{
+    [Cmdlet(VerbsLifecycle.Request, "ContainerImage",
+            DefaultParameterSetName = CommonParameterSetNames.Default)]
+    [OutputType(typeof(ContainerListResponse))]
+    [Alias("Pull-ContainerImage")]
+    public class RequestContainerImage : DkrCmdlet
+    {
+        private string _statusUpToDate = "Status: Image is up to date for ";
+        private string _statusDownloadedNewer = "Status: Downloaded newer image for ";
+
+        [Parameter(ParameterSetName = CommonParameterSetNames.Default,
+            ValueFromPipeline = true,
+            Mandatory = true,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string Repository { get; set; }
+
+        [Parameter(ParameterSetName = CommonParameterSetNames.Default,
+            Position = 1)]
+        [ValidateNotNullOrEmpty]
+        public string Tag { get; set; }
+
+        [Parameter]
+        public SwitchParameter All { get; set; }
+
+        [Parameter]
+        public AuthConfig Authorization { get; set; }
+
+        #region Overrides
+        protected override async Task ProcessRecordAsync()
+        {
+            string repoTag = null;
+            bool failed = false;
+            var pullTask = DkrClient.Images.PullImageAsync(new ImagesPullParameters() { All = All, Parent = Repository, Tag = Tag, }, Authorization);
+            var messageWriter = new JsonMessageWriter(this);
+
+            using (var pullStream = await pullTask)
+            {
+                // ReadLineAsync is not cancellable without closing the whole stream, so register a callback to do just that.
+                using (CmdletCancellationToken.Register(() => pullStream.Close()))
+                using (var pullReader = new StreamReader(pullStream, new UTF8Encoding(false)))
```

Comment:
I still wish this was done at the Docker.DotNet level using an `IProgress<T> where T : JsonMessage`  and then the pull task is just a task and its "stream" is sent as progress notifications. Makes 100% more sense to me than this model.